src/source/Ice/TurnConnection.c

```diff
@@ -7,6 +7,48 @@
 extern StateMachineState TURN_CONNECTION_STATE_MACHINE_STATES[];
 extern UINT32 TURN_CONNECTION_STATE_MACHINE_STATE_COUNT;
 
+// On success, ppHostname is allocated and must be freed by the caller
+STATUS getHostnameFromUrl(PCHAR url, PCHAR* ppHostname)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    PCHAR urlStart = NULL;
+    PCHAR urlEnd = NULL;
+    UINT32 hostnameLen = 0;
+
+    CHK(url != NULL && ppHostname != NULL, STATUS_NULL_ARG);
+
+    // Skip protocol prefix (turn: or turns:) if present
+    urlStart = STRSTR(url, "://");
+    if (urlStart != NULL) {
+        urlStart += 3; // Skip "://"
+    } else {
+        // If no protocol prefix, use the entire URL
+        urlStart = url;
+    }
+
+    // Find port separator if it exists
+    urlEnd = STRCHR(urlStart, ':');
```

Comment:
Should we check if a '?' parameter exists without a port number? queryStart = STRCHR(urlStart, '?');

Reply:
Improved to support URLs without port.